konrad/cloud.py

```diff
@@ -97,27 +100,110 @@ def __init__(self, numlevels=200, num_lw_bands=16, num_sw_bands=14,
 
         self.longwave_optical_thickness_due_to_cloud = get_waveband_data_array(
                 lw_optical_thickness, numlevels=numlevels,
-                numbands=num_lw_bands, sw=False)
+                numbands=self.num_longwave_bands, sw=False)
 
         self.cloud_forward_scattering_fraction = get_waveband_data_array(
                 forward_scattering_fraction, numlevels=numlevels,
-                numbands=num_sw_bands)
+                numbands=self.num_shortwave_bands)
 
         self.cloud_asymmetry_parameter = get_waveband_data_array(
                 asymmetry_parameter, numlevels=numlevels,
-                numbands=num_sw_bands)
+                numbands=self.num_shortwave_bands)
 
         self.shortwave_optical_thickness_due_to_cloud = \
             get_waveband_data_array(sw_optical_thickness,
                                     numlevels=numlevels,
-                                    numbands=num_sw_bands)
+                                    numbands=self.num_shortwave_bands)
 
         self.single_scattering_albedo_due_to_cloud = get_waveband_data_array(
                 single_scattering_albedo, numlevels=numlevels,
-                numbands=num_sw_bands)
+                numbands=self.num_shortwave_bands)
+
+    def interpolation_function(self, cloud_parameter, z):
+        """ Calculate the interpolation function, to be used to keep the
+        cloud at a constant thickness [m] and attached to a normalisation
+        level (self._norm_level). A separate interpolation function is required
+        for each cloud parameter that needs to be interpolated.
+
+        Parameters:
+            cloud_parameter (DataArray): property to be interpolation
+            z (ndarray): height array, on which the interpolation is performed
+        Returns:
+            scipy.interpolate.interpolate.interp1d
+        """
+        normed_height = z - self._norm_level
+        interpolation_f = interp1d(
+            normed_height,
+            cloud_parameter.values,
+            axis=0,
+            fill_value='extrapolate',
+        )
+        return interpolation_f
+
+    def shift_property(self, cloud_parameter, interpolation_f, z, norm_new):
+        """Shift the cloud area fraction according to a normalisation level.
+        Maintain the thickness of the cloud [m].
+
+        Parameters:
+            cloud_parameter (DataArray): cloud property to be shifted
+            interpolation_f (scipy.interpolate.interpolate.interp1d):
+                interpolation object calculated by interpolation_function
+            z (ndarray): height array [m]
+            norm_new (int / float): normalisation height [m]
+
+        Returns:
+            DataArray: shifted cloud property
+        """
+        # Move the cloud to the new normalisation level, if there is one.
+        # Otherwise keep the cloud where it is.
```

Comment:
It is just a matter of taste, but I tend to comment if-else-statements as follows:
```python
if norm_new is not np.nan:
    # Move the cloud to the new normalisation level, if there is one.
    cloud_parameter.values = interpolation_f(z - norm_new)
else:
    # Otherwise keep the cloud where it is.
    cloud_parameter.values = interpolation_f(z - self._norm_level)
```